src/internal/collection/types.go

```diff
@@ -93,6 +93,10 @@ type PostgresReadWriteCollection interface {
 	// a large number of rows. Consider using a read-only collection if possible
 	GetByIndex(index *Index, indexVal string, val proto.Message, opts *Options, f func(string) error) error
 
+	// GetUniqueByIndex is identical to GetByIndex except it is an error if
```

Comment:
This seems like something that would be better as an option on `GetByIndex` or a wrapper, rather than an additional function on the interface.

Reply:
That's basically how it's implemented internally, but I'd prefer not to have bool parameters in a function, they are very hard to read.  A wrapper could work, but it sacrifices the ergonomics of using the collection in a similar way.

Reply:
This could also just be an option on `GetByIndex`. My concern with functions like this is interface bloat, but I'm not really strongly against this approach.